### DISCUSSION TITLE
Inject a service into custom node view

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#934] [Q&A]
Hi,
   I create a custom data service, I want to inject it into my custom node view. I find the service is always undefined in the node view, I debug and find the node view is constructed first and then construct my data service.
![image](https://user-images.githubusercontent.com/4537809/221495633-8e8492fa-7e4a-499d-9e8b-ae552380f721.png)
![image](https://user-images.githubusercontent.com/4537809/221495730-d541e9fb-84d9-4566-bdbe-564527bed9af.png)
![image](https://user-images.githubusercontent.com/4537809/221496868-8f1ff4d6-f6db-40d8-9761-79b23f999b99.png)
is it possible to create my custom service for node view? and where is the proper place for my data service, I want to invoke the stop method in service when the file is closed.

**Comment by CamilleLetavernier** [comment]
Hey,

If you need to access the service directly in the constructor, you should use constructor-injection instead of field-injection. Fields will be injected after the constructor is invoked, so they won't exist inside of the constructor.

Constructor injection looks like this:

```ts
constructor(@inject(DataMonitorService) protected dataService: DataMonitorService){
   // ... Constructor code that accesses dataService...
}
````

**Reply by wss29** [reply]
Hi,
    Oh yes, I misunderstood the order of injection, It is ok now.
another problem I want to invoke the stop method of my service when the diagram was closed, but I look at these demo code like workflow. It seems that only a di.config.ts,  where can I get the close event of the diagram? or follow error will always refresh output page
![image](https://user-images.githubusercontent.com/4537809/221528922-15bd8a72-8a75-411c-9c03-0cc89b7b092f.png)
